src/Rpc/NodeGuardService.cs

```diff
@@ -1033,33 +1033,41 @@ private WITHDRAWAL_REQUEST_STATUS GetStatus(WalletWithdrawalRequestStatus status
         };
     }
 
-    private GetWithdrawalsRequestStatusResponse GetWithdrawalsRequestStatusResponse(List<WalletWithdrawalRequest>? withdrawalRequests)
+    private async Task<GetWithdrawalsRequestStatusResponse> GetWithdrawalsRequestStatusResponse(List<WalletWithdrawalRequest> withdrawalRequests)
```

Comment:
dumb question: why does this need to change to async task?

Reply:
`Task` is always needed as a return parameter to await for an async call. You can't for example do `await GetSomething()` if the signature is `void` instead of `Task`

Reply:
weren't we already performing an async task when querying the db?

Reply:
not inside the function `GetWithdrawalsRequestStatusResponse`